Sources/SPTDataLoaderService.m

```diff
@@ -237,6 +237,8 @@ - (void)performRequest:(SPTDataLoaderRequest *)request
     if (request.backgroundPolicy == SPTDataLoaderRequestBackgroundPolicyAlways) {
         task = [self.session downloadTaskWithRequest:urlRequest];
     } else {
+        NSCachedURLResponse *cachedResponse = [self.session.configuration.URLCache cachedResponseForRequest:urlRequest];
+        request.isCached = cachedResponse != nil;
```

Comment:
Is there no other way to know that a response came from the cache? It feels a little weird to mutate the _request_ for this, no? Wouldn't it make more sense as a property on the response object? `cameFromCache` - something like that?

Reply:
I didn’t find a better way to find out if a response may be coming from the cache. It seems at this point in code the response isn’t available yet, that’s why the request is mutated. Moreover the request is used in request accounting reporting therefore it’s convenient to store the property in the request object. We need this property to avoid overreporting network traffic in request accounting messages.

Reply:
Have you verified that the right value comes back in the delegate callbacks? IIRC the request is `copy`ed at some point.

Reply:
We should probably use the async `getCachedResponseForDataTask:completionHandler:` method, and if we get a hit, we can short-circuit even pushing the data through the URL session and just fire the delegate callbacks manually. Maybe easier said than done.

Reply:
Do you have examples of using `getCachedResponseForDataTask`? I can't seem to find any.